modules/fluid_properties/src/userobjects/SalineFluidProperties.C

```diff
@@ -0,0 +1,173 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+//
+#include "SalineFluidProperties.h"
+
+registerMooseObject("FluidPropertiesApp", SalineFluidProperties);
+
+InputParameters
+SalineFluidProperties::validParams()
+{
+  InputParameters params = SinglePhaseFluidProperties::validParams();
+  params.addRequiredParam<std::string>(
+      "comp_name", "The name of the components in the salt (e.g., 'LiF-NaF-KF')");
+  params.addRequiredParam<std::string>(
+      "comp_val", "The mole fraction of each salt component (e.g., '0.5-0.1-0.4)");
+  params.addRequiredParam<std::string>(
+      "prop_def",
+      "Definition of a fluid property file, which must be a file path to the "
+      "comma-separated data matching the Saline format.");
+  return params;
+}
+
+// Input parser for salt component names
+std::vector<std::string>
+splitString_str(std::string str)
+{
+  std::string delimiter = "-";
+  std::string tmpstr = str;
+
+  std::vector<std::string> list;
+  size_t pos = 0;
+  std::string token;
+  while ((pos = tmpstr.find(delimiter)) != std::string::npos)
+  {
+    token = tmpstr.substr(0, pos);
+    list.push_back(token);
+    tmpstr.erase(0, pos + delimiter.length());
+  }
+  list.push_back(tmpstr);
+  return list;
+}
+
+// Input parser for salt component mole fractions
+std::vector<Real>
+splitString_float(std::string str)
+{
+  std::vector<std::string> list = splitString_str(str);
+  std::vector<Real> res;
+  for (std::string s : list)
+    res.push_back(stof(s));
+  return res;
+}
+
+SalineFluidProperties::SalineFluidProperties(const InputParameters & parameters)
+  : SinglePhaseFluidProperties(parameters)
+{
+
+#ifdef SALINEAPP_H
+
+  std::string propDef = getParam<std::string>("prop_def");
+  _d.load(propDef);
+  bool success = _tp.initialize(&_d);
+  if (!success)
+    mooseError("The initialize of the Saline interface has failed");
+  std::string name = getParam<std::string>("comp_name");
+  std::string comp = getParam<std::string>("comp_val");
+  std::vector<std::string> nameList = splitString_str(name);
+  std::vector<Real> valList = splitString_float(comp);
+  Real mole_sum = 0.0;
+  for (Real val : valList)
+    mole_sum = mole_sum + val;
+  if (abs(mole_sum - 1.0) > 1e-6)
```

Comment:
```suggestion
  if (std::abs(mole_sum - 1.0) > 1e-6)
```

we always explicitly specify the namespace for the math functions